code/API_definitions/BYON-Notification-Channel.yaml

```diff
@@ -2,69 +2,135 @@ openapi: 3.0.0
 info:
   title: Bring Your Own Number (BYON) Notification Channel APIs
   description: This API provide REST API for a client, browser or native application, to establish notification channel to receive asynchronous notifications from MNO's IMS Network.
-  version: 0.1.0
+  version: 0.1.1
   contact:
     email: contact@domain.com
   license:
     name: Apache 2.0
     url: http://www.apache.org/licenses/LICENSE-2.0.html
 servers:
-- url: '{apiRoot}/notificationchannel/{apiVersion}/{deviceId}'
-  description: APIs to create Notification Channels
+- url: '{apiRoot}/notificationchannel/{apiVersion}'
```

Comment:
Notification channels created will be associated with the device instance from which the WebRTC sessions are being initiated. Removal of {deviceId} from the URI will lead to WebRTC gateway not associating channels to the deviceId. Please share your rationale for removal of deviceId, as stated our preference and recommendation is to retain it so that the network can maintain the device instance to notification channel association.

Reply:
We understand your preference to retain the deviceId in the URI for associating notification channels with WebRTC sessions. However, we propose removing it as a mandatory parameter for the following reasons:

**ClientId Sufficiency for Gateway Channels:** When using a WebRTC gateway that supports notification channels, the clientId can uniquely identify the device instance initiating the session. This eliminates the need for an additional deviceId parameter.

**External Service Flexibility:**  For notification channels managed by external services,  the notification channelId provided during WebRTC session creation already establishes the association. Introducing deviceId adds unnecessary complexity.

**Multi-Device Notifications:**  Certain use cases, such as a calling service, might require a single notification channel to receive notifications for multiple devices. Removing the mandatory deviceId allows developers to implement this scenario by using a common Webhook notification channel. This channel would then deliver notifications to a central web service, which would then handle routing them to the specific devices.

Therefore, we recommend making deviceId an optional parameter to provide flexibility and cater to diverse notification channel implementations.